src/monsters.cpp

```diff
@@ -722,77 +719,76 @@ bool Monsters::loadMonster(const std::string& file, const std::string& monsterNa
 		return false;
 	}
 
-	if (reloading) {
-		mType = getMonsterType(monsterName);
-		if (mType != nullptr) {
-			new_mType = false;
-			mType->reset();
-		}
-	}
-
-	if (new_mType) {
-		mType = &monsters[asLowerCaseString(monsterName)];
-	}
-
-	mType->name = attr.as_string();
+	MonsterType mType;
+	mType.name = attr.as_string();
 
 	if ((attr = monsterNode.attribute("nameDescription"))) {
-		mType->nameDescription = attr.as_string();
+		mType.nameDescription = attr.as_string();
 	} else {
-		mType->nameDescription = "a " + mType->name;
-		toLowerCaseString(mType->nameDescription);
+		mType.nameDescription = "a " + asLowerCaseString(mType.name);
 	}
 
 	if ((attr = monsterNode.attribute("race"))) {
 		std::string tmpStrValue = asLowerCaseString(attr.as_string());
 		uint16_t tmpInt = pugi::cast<uint16_t>(attr.value());
 		if (tmpStrValue == "venom" || tmpInt == 1) {
-			mType->race = RACE_VENOM;
+			mType.race = RACE_VENOM;
 		} else if (tmpStrValue == "blood" || tmpInt == 2) {
-			mType->race = RACE_BLOOD;
+			mType.race = RACE_BLOOD;
 		} else if (tmpStrValue == "undead" || tmpInt == 3) {
-			mType->race = RACE_UNDEAD;
+			mType.race = RACE_UNDEAD;
 		} else if (tmpStrValue == "fire" || tmpInt == 4) {
-			mType->race = RACE_FIRE;
+			mType.race = RACE_FIRE;
 		} else if (tmpStrValue == "energy" || tmpInt == 5) {
-			mType->race = RACE_ENERGY;
+			mType.race = RACE_ENERGY;
 		} else {
 			std::cout << "[Warning - Monsters::loadMonster] Unknown race type " << attr.as_string() << ". " << file << std::endl;
 		}
 	}
 
 	if ((attr = monsterNode.attribute("experience"))) {
-		mType->experience = pugi::cast<uint64_t>(attr.value());
+		mType.experience = pugi::cast<uint64_t>(attr.value());
 	}
 
 	if ((attr = monsterNode.attribute("speed"))) {
-		mType->baseSpeed = pugi::cast<int32_t>(attr.value());
+		int32_t baseSpeed = pugi::cast<int32_t>(attr.value());
+		if (baseSpeed >= 0) {
+			mType.baseSpeed = baseSpeed;
+		} else {
+			std::cout << "[Error - Monsters::loadMonster] Invalid speed value " << mType.baseSpeed << ". " << file << std::endl;
+			return false;
+		}
 	}
 
 	if ((attr = monsterNode.attribute("manacost"))) {
-		mType->manaCost = pugi::cast<uint32_t>(attr.value());
+		mType.manaCost = pugi::cast<uint32_t>(attr.value());
 	}
 
 	if ((attr = monsterNode.attribute("skull"))) {
-		mType->skull = getSkullType(attr.as_string());
+		mType.skull = getSkullType(attr.as_string());
 	}
 
 	if ((attr = monsterNode.attribute("script"))) {
-		monsterScriptList.emplace_back(mType, attr.as_string());
+		monsterScriptList.emplace_back(&mType, attr.as_string());
```

Comment:
Is there any guarantee that this will point to the object once it is moved into the monsters map, and that the monster object will retain the same position in memory, given that monsters is a std::map, which is usually implemented as a red-black tree, which can be re-balanced?